### DISCUSSION TITLE
Using Faker for anonymizing data.

### DISCUSSION THREAD

**Discussion by nautics889** [sinaptik-ai/pandas-ai#827] [Ideas]
At first, thanks to @gventuri for merging https://github.com/gventuri/pandas-ai/pull/826 so soon. 
These patch contained some refactoring of [`Anonymizer`](https://github.com/gventuri/pandas-ai/blob/0b7cecc469f5086f966256173b302aeeb03f0fb2/pandasai/helpers/anonymizer.py#L13) class, and when I was working on it, i've found it quite strange that we're generating mocked data ourself (like [e-mails](https://github.com/gventuri/pandas-ai/blob/0b7cecc469f5086f966256173b302aeeb03f0fb2/pandasai/helpers/anonymizer.py#L54), [card numbers](https://github.com/gventuri/pandas-ai/blob/0b7cecc469f5086f966256173b302aeeb03f0fb2/pandasai/helpers/anonymizer.py#L92), etc). I mean, instead we can use [Faker](https://pypi.org/project/Faker/). 
Moreover, we already have Faker [in the project dependencies list](https://github.com/gventuri/pandas-ai/blob/0b7cecc469f5086f966256173b302aeeb03f0fb2/pyproject.toml#L21) (and not development one, btw).

1. Shouldn't we use Faker in Anonymizer, for generating anonymous data for PandasAI requests?
2. Should we generate some use Faker for our tests to create fixtures? (I bet we should)

___

P. S. decided to create a discussion topic rather than an issue, I feel it more like suggestion, not a feature request

**Comment by gventuri** [comment]
Hi @nautics889, thanks a lot for the contributions :)

1) Yeah, I think we weren't using faker yet when we wrote the anonymizer, but I'd totally agree we should use faker instead!
2) Agreed on this point as well!
